packages/react/README.md

```diff
@@ -18,6 +18,124 @@ This package contains React hooks for building Solana apps.
 
 ## Hooks
 
+### `useWalletAccountMessageSigner(uiWalletAccount)`
+
+Given a `UiWalletAccount`, this hook returns an object that conforms to the `MessageModifyingSigner` interface of `@solana/signers`.
+
+#### Example
+
+```tsx
+import { useWalletAccountMessageSigner } from '@solana/react';
+import { createSignableMessage } from '@solana/signers';
+
+function SignMessageButton({ account, text }) {
+    const messageSigner = useWalletAccountMessageSigner(account);
+    return (
+        <button
+            onClick={async () => {
+                try {
+                    const signableMessage = createSignableMessage(text);
+                    const [signedMessage] = await messageSigner.modifyAndSignMessages([signableMessage]);
+                    const messageWasModified = signableMessage.content !== signedMessage.content;
+                    const signatureBytes = signedMessage.signatures[messageSigner.address];
+                    window.alert(
+                        `Signature bytes: ${signatureBytes.toString()}${
+                            messageWasModified ? ' (message was modified)' : ''
+                        }`,
+                    );
+                } catch (e) {
+                    console.error('Failed to sign message', e);
+                }
+            }}
+        >
+            Sign Message: {text}
+        </button>
+    );
+}
+```
+
+> [!NOTE]
+> The type `MessageModifyingSigner` is returned from this hook instead of `MessageSigner` or `MessagePartialSigner`. This is a conservative assumption based on the fact that your application can not control whether or not the wallet will modify the message before signing it.
```

Comment:
Makes sense to me. Even if we made the hook generic and required the type of message signer to be piped through wallet standard, it can easily be spoofed.